src/network/networkd-address.c

```diff
@@ -1911,11 +1917,33 @@ static int address_section_verify(Address *address) {
 }
 
 void network_drop_invalid_addresses(Network *network) {
+        _cleanup_set_free_ Set *addresses = NULL;
         Address *address;
+        int r;
 
         assert(network);
 
-        ORDERED_HASHMAP_FOREACH(address, network->addresses_by_section)
-                if (address_section_verify(address) < 0)
+        ORDERED_HASHMAP_FOREACH(address, network->addresses_by_section) {
+                Address *dup;
+
+                if (address_section_verify(address) < 0) {
                         address_free(address);
+                        continue;
+                }
+
+                /* Always use the setting specified later. */
+                dup = set_remove(addresses, address);
+                if (dup) {
+                        log_warning("%s: Duplicated address is specified, dropping address specified at line %u.",
+                                    dup->section->filename, dup->section->line);
+                        address_free(dup);
```

Comment:
I would add a comment here saying that `address_free` removes the address from `addresses_by_section` as well since that's non-trivial to infer from the code here.